corpus/declarations.txt

```diff
@@ -227,146 +271,182 @@ func g1[T, U any, V interface{}, W Foo[Bar[T]]](a Foo[T]) {}
     (identifier)
     (parameter_list)
     (parameter_list
-      (parameter_declaration (type_identifier))
-      (parameter_declaration (type_identifier)))
+      (parameter_declaration
+        (type_identifier))
+      (parameter_declaration
+        (type_identifier)))
     (block))
   (function_declaration
     (identifier)
     (parameter_list)
     (parameter_list
-      (parameter_declaration (identifier) (type_identifier))
-      (parameter_declaration (identifier) (type_identifier)))
+      (parameter_declaration
+        (identifier)
+        (type_identifier))
+      (parameter_declaration
+        (identifier)
+        (type_identifier)))
     (block))
   (function_declaration
     (identifier)
     (type_parameter_list
-      (parameter_declaration
+      (type_parameter_declaration
         (identifier)
         (identifier)
-        (type_identifier))
-      (parameter_declaration
+        (constraint_elem
+          (constraint_term
+            (type_identifier))))
+      (type_parameter_declaration
         (identifier)
-        (interface_type))
-      (parameter_declaration
+        (constraint_elem
+          (constraint_term
+            (type_identifier))))
+      (ERROR)
```

Comment:
Here we've introduced an `ERROR` node. Maybe the `test --update` command should issue a warning in that case.

Reply:
Sorry, it wasn't ready for review yet; I'm still guessing at the correct test process (I added my guess to the README). The --update flag is reformatting the file in a way that makes it hard to see the actual change; I wondered if that was due to version skew, and pushed to the branch in the hope that Actions would give me the canonical result.

How should I execute and update the tests to avoid the spurious diffs?

Reply:
In my work, I've typically just updated the tests by hand. I often do it before making the corresponding grammar change, just as a way of thinking concretely about the syntax tree that I want to see. Admittedly, it can sometimes be tedious to do when a given change affects many different tests.

The `--update` flag is a very cool feature that was added recently in a PR; I think we just need to refine its behavior a bit. It should definitely not just silently insert `ERROR` nodes into the expectations. It should probably refuse to write any files if an ERROR has been introduced in an existing test.

Reply:
I opened an issue to track that: https://github.com/tree-sitter/tree-sitter/issues/1691.